lib/sles4sap_publiccloud.pm

```diff
@@ -468,8 +468,20 @@ sub start_hana {
 sub cleanup_resource {
     my ($self, %args) = @_;
     my $timeout = bmwqemu::scale_timeout($args{timeout} // 300);
-
-    $self->run_cmd(cmd => 'crm resource cleanup');
+    my @errors = (
+        qr/Aborting because no messages received in \d+ seconds/,
+        qr/Error performing operation: Timeout occurred/
+    );
+    my $retry = '3';
+    my $out = undef;
+
+    # Retry when running crm command with issue, refer to TEAM-10483
+    while ($retry) {
+        $out = $self->run_cmd(cmd => 'crm resource cleanup', proceed_on_failure => 1);
+        last if (!grep { $out =~ /$_/ } @errors);
+        $retry--;
+    }
+    die "Clean up resource failed" if ($retry == 0);
```

Comment:
No need for double quotes. Interpolation is not needed here. 

You can on the other hand include output from the command ion the `die` message:
```
die "Clean up resource failed:\n$out" if ($retry == 0);
```